components/encryption/src/io.rs

```diff
@@ -514,6 +612,15 @@ impl CrypterCore {
     }
 
     pub fn do_crypter(&mut self, buf: &[u8]) -> IoResult<&[u8]> {
+        assert_eq!(
```

Comment:
maybe `debug_assert_eq!`

Reply:
If this happens in a production cluster, the data will be corrupted. I think panic here instead will be better than writing wrong data.